### ISSUE TITLE
Plugin v0.4.10 on Unity 4.6.4 made the camera upside down on mobile device.

### ISSUE THREAD

**Issue by LegACy99** [googlesamples/cardboard-unity#29]
Running an Android app that's built using plugin v0.4.10 on Unity 4 (I tried on both Unity 4.5 and 4.6.4f1 free edition) made the camera go upside down. Here's a screenshot from the demo scene, notice the tiled floor is on the top side of the screen.

![reversed](https://cloud.githubusercontent.com/assets/1537053/7215750/3763fb88-e60f-11e4-8d3e-a67a4c30a53d.png)

The camera renders fine on the Editor, but on a mobile device (I tried it with a Nexus 5) it's upside down. When it's built with plugin v0.4.9 instead of v0.4.10, the camera renders normally on a mobile device.

**Comment by smdol** [comment]
Can you say what your phone model and Android version are?

**Comment by smdol** [comment]
That's a bug in the 0.4.10 that was thought to be fixed, but apparently the cause is not as understood as was thought.  In my tests, this scene came out the correct way (floor down).  Looking in to it...

**Comment by smdol** [comment]
Have you tried Unity 5 as well?

**Comment by smdol** [comment]
If you turn off nativeDistortionCorrection, it should also go back to normal.

**Comment by smdol** [comment]
OK, I got this to happen on my phone too.  Sad, because it was supposed to be fixed before 0.4.10 was released.  

So if you are using Unity 4 Free, turn off nativeDistortionCorrection on the Cardboard object to restore the behavior to what it was in 0.4.9.

**Comment by LegACy99** [comment]
Heh, I wasn't expecting any replies until at least monday =)

So yeah, disabing nativeDistortionCorrection does work and the camera renders just fine now. But umm, I thought the distortionCorrection is sorta important in VR?

**Comment by smdol** [comment]
Yep, and that's why I'm bummed you ran  into this.  Although, if Unity 5 is
a possibility, give it a shot.  It is free and supports distortion
correction already.

**Comment by LegACy99** [comment]
Unfortunately with Unity 5 I'll have to rework the lighting and shader stuff, and I cannot afford that hassle right now. I suppose I'll stick with 0.4.9 for now.

Thanks for the help btw.

**Comment by LegACy99** [comment]
Okay I hope SDK 0.5 fixes this =D
Will report back once I have free time (AKA when today's I/O is done).

**Comment by smdol** [comment]
Yep, it's a bug in 0.4.10 for Unity Free.  It is not there in 0.4.9 or in
0.5.

-- Scott

**Comment by LegACy99** [comment]
Yep it's fixed. And I didn't see any other problem after upgrading to 0.5 too. I'll close this issue =)

**Comment by Guhanesh** [comment]
I am experiencing this issue wit Cardboard SDK for unity (Ver 0.7.0)  with Unity 5.3.1
Any solutions for this one?

**Comment by smdol** [comment]
We completely removed the code path that caused the original issue, almost a year ago, so the problem you are having must be a new one even if it looks similar -- a new cause anyway.

Can you open a new issue, and add some more details?  (Phone model, screenshot, is the default orientation set to Landscape left)

**Comment by Guhanesh** [comment]
Sorry setting orientation to landscape left solves the problem

--- Original Message ---
